app/frontend/wallet/keypass-json.js

```diff
@@ -102,23 +110,29 @@ async function isWalletExportEncrypted(walletExportObj) {
 }
 
 async function importWalletSecret(walletExportObj, password) {
-  const {passwordHash, walletSecret} = parseWalletExportObj(walletExportObj)
+  const {passwordHash, walletSecretDef} = parseWalletExportObj(walletExportObj)
 
   const isPasswordVerified = await verifyPassword(password, passwordHash)
   if (!isPasswordVerified) {
     throw Error('Wrong password')
   }
 
   if (!password) {
-    return walletSecret
+    return walletSecretDef
   }
 
-  return decryptWalletSecret(walletSecret, password)
+  walletSecretDef.rootSecret = decryptWalletSecret(walletSecretDef.rootSecret, password)
+
+  return walletSecretDef
 }
 
-async function exportWalletSecret(walletSecret, password, walletName) {
+async function exportWalletSecret(walletSecret, password, walletName, derivationScheme) {
```

Comment:
I would refactor this function to take a `walletSecretDef` object and rename it `exportWalletSecretDef` for the sake of consistency